contracts/allowlist/OperatorAllowlistUpgradeable.sol

```diff
@@ -0,0 +1,189 @@
+    }
+
+    ///     =====  External functions  =====
+
+    /**
+     * @notice Add a target address to Allowlist
+     * @param addressTargets the addresses to be added to the allowlist
+     */
+    function addAddressToAllowlist(address[] calldata addressTargets) external onlyRole(REGISTRAR_ROLE) {
+        for (uint256 i; i < addressTargets.length; i++) {
+            addressAllowlist[addressTargets[i]] = true;
+            emit AddressAllowlistChanged(addressTargets[i], true);
+        }
+    }
+
+    /**
+     * @notice Remove a target address from Allowlist
+     * @param addressTargets the addresses to be removed from the allowlist
+     */
+    function removeAddressFromAllowlist(address[] calldata addressTargets) external onlyRole(REGISTRAR_ROLE) {
+        for (uint256 i; i < addressTargets.length; i++) {
+            delete addressAllowlist[addressTargets[i]];
+            emit AddressAllowlistChanged(addressTargets[i], false);
+        }
+    }
+
+    /**
+     * @notice Add a smart contract wallet to the Allowlist.
+     * This will allowlist the proxy and implementation contract pair.
+     * First, the bytecode of the proxy is added to the bytecode allowlist.
+     * Second, the implementation address stored in the proxy is stored in the
+     * implementation address allowlist.
+     * @param walletAddr the wallet address to be added to the allowlist
+     */
+    function addWalletToAllowlist(address walletAddr) external onlyRole(REGISTRAR_ROLE) {
+        // get bytecode of wallet
+        bytes32 codeHash;
+        assembly {
+            codeHash := extcodehash(walletAddr)
+        }
+        bytecodeAllowlist[codeHash] = true;
+        // get address of wallet module
+        address impl = IProxy(walletAddr).PROXY_getImplementation();
+        addressImplementationAllowlist[impl] = true;
+
+        emit WalletAllowlistChanged(codeHash, walletAddr, true);
+    }
+
+    /**
+     * @notice Remove  a smart contract wallet from the Allowlist
+     * This will remove the proxy bytecode hash and implementation contract address pair from the allowlist
+     * @param walletAddr the wallet address to be removed from the allowlist
+     */
+    function removeWalletFromAllowlist(address walletAddr) external onlyRole(REGISTRAR_ROLE) {
+        // get bytecode of wallet
+        bytes32 codeHash;
+        assembly {
+            codeHash := extcodehash(walletAddr)
+        }
+        delete bytecodeAllowlist[codeHash];
+        // get address of wallet module
+        address impl = IProxy(walletAddr).PROXY_getImplementation();
+        delete addressImplementationAllowlist[impl];
+
+        emit WalletAllowlistChanged(codeHash, walletAddr, false);
+    }
+
+    /**
+     * @notice Allows admin to grant `user` `REGISTRAR_ROLE` role
+     * @param user the address that `REGISTRAR_ROLE` will be granted to
+     */
+    function grantRegistrarRole(address user) external onlyRole(DEFAULT_ADMIN_ROLE) {
```

Comment:
This function is not needed. It is available in the AccessControl Open Zeppelin contract.